jupyter-pytorch-full/rockcraft.yaml

```diff
@@ -141,20 +148,47 @@ parts:
       "${CONDA_BIN}/conda" clean -a -f -y
 
       # Install the jupyter python requirements
+      echo "jupyterlab ==${JUPYTERLAB_VERSION}" >> ${CONDA_DIR}/conda-meta/pinned
+      echo "notebook ==${JUPYTER_VERSION}" >> ${CONDA_DIR}/conda-meta/pinned
+      "${CONDA_BIN}/conda" install -y -q \
+        jupyterlab==${JUPYTERLAB_VERSION} \
+        notebook==${JUPYTER_VERSION}
       cp ${CRAFT_PART_SRC}/components/example-notebook-servers/jupyter/requirements.txt jupyter-requirements.txt
       "${CONDA_BIN}/pip" install --no-cache-dir -r jupyter-requirements.txt
 
       # Generate a jupyter lab config
       "${CONDA_BIN}/jupyter" lab --generate-config
+      "${CONDA_BIN}/jupyter" notebook --generate-config
+      "${CONDA_BIN}/jupyter" labextension disable "@jupyterlab/apputils-extension:announcements"
 
       # Install the jupyter-pytorch cpu requirements
-      cp $CRAFT_PART_SRC/components/example-notebook-servers/jupyter-pytorch/cpu-requirements.txt cpu-requirements.txt
-      "${CONDA_BIN}/pip" install --no-cache-dir -r cpu-requirements.txt
+      "${CONDA_BIN}/pip" install --quiet --no-cache-dir --index-url https://download.pytorch.org/whl/cpu --extra-index-url https://pypi.org/simple \
+        torch==${PYTORCH_VERSION} \
+        torchvision==${TORCHVISION_VERSION} \
+        torchaudio==${TORCHAUDIO_VERSION}
+      cp $CRAFT_PART_SRC/components/example-notebook-servers/jupyter-pytorch/requirements.txt requirements.txt
+      "${CONDA_BIN}/pip" install --no-cache-dir -r requirements.txt
 
       # Install the jupyter-pytorch-full cpu requirements
+      "${CONDA_BIN}/mamba" install -y -q \
+        bokeh==3.2.2 \
+        cloudpickle==2.2.1 \
+        dill==0.3.7 \
+        ipympl==0.9.3 \
+        matplotlib==3.8.0 \
+        pandas==2.1.1 \
+        scikit-image==0.22.0 \
+        scikit-learn==1.3.1 \
+        scipy==1.11.3 \
+        seaborn==0.13.0 \
+        xgboost==1.7.6
       cp $CRAFT_PART_SRC/components/example-notebook-servers/jupyter-pytorch-full/requirements.txt requirements.txt
       "${CONDA_BIN}/pip" install --no-cache-dir -r requirements.txt
 
+      # Install mlflow
```

Comment:
Should we have a comment why we add MLflow here? Since this is something that upstream doesn't have. This could also be a link to an issue if you think it fits better.

Reply:
added comment